lib/full-site-editing/template-loader.php

```diff
@@ -266,3 +266,28 @@ function gutenberg_resolve_template_for_new_post( $wp_query ) {
 		$wp_query->set( 'post_status', 'auto-draft' );
 	}
 }
+
+/**
+ * Redirect the edit links for templates to the site editor.
+ *
+ * @param string $link    The original link.
+ * @param int    $post_id The custom post id.
+ */
+function gutenberg_get_edit_template_link( $link, $post_id ) {
+	$post = get_post( $post_id );
+
+	if ( 'wp_template' !== $post->post_type && 'wp_template_part' !== $post->post_type ) {
+		return $link;
+	}
+
+	$template = _build_block_template_result_from_post( $post );
+
+	if ( ! $template ) {
```

Comment:
I checked `_build_block_template_result_from_post` and I wonder if we ever get $template `false` from it.

Reply:
Fixes in 21656838ab186acef34363369cdaf721a62b36a0. Thanks for catching this, Andrei.